aspnetcore/mvc/models/model-binding/samples/6.0/ModelBindingSample/Snippets/Program.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Metadata;
+
+namespace ModelBindingSample.Snippets;
+
+public static class Program
+{
+    public static void AddValueProviderFactory(WebApplicationBuilder builder)
+    {
+        // <snippet_AddValueProviderFactory>
+        builder.Services.AddControllers(options =>
+        {
+            options.ValueProviderFactories.Add(new CookieValueProviderFactory());
+        });
+        // </snippet_AddValueProviderFactory>
+    }
+
+    public static void ReplaceQueryStringValueProviderFactory(WebApplicationBuilder builder)
+    {
+        // <snippet_ReplaceQueryStringValueProviderFactory>
+        builder.Services.AddControllers(options =>
+        {
+            var index = options.ValueProviderFactories.IndexOf(
+                options.ValueProviderFactories.OfType<QueryStringValueProviderFactory>().Single());
```

Comment:
```suggestion
             options.ValueProviderFactories.OfType<QueryStringValueProviderFactory>().Single());
```
This suggestion will remove the horizontal scroll bar.